gst/nnstreamer/tensor_source/tensor_src_iio.c

```diff
@@ -0,0 +1,384 @@
+/**
+ * GStreamer Tensor_Source_IIO
+ * Copyright (C) 2005 Thomas Vander Stichele <thomas@apestaart.org>
+ * Copyright (C) 2005 Ronald S. Bultje <rbultje@ronald.bitfreak.net>
+ * Copyright (C) 2019 Parichay Kapoor <pk.kapoor@samsung.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ */
+/**
+ * @file	tensor_src_iio.c
+ * @date	27 Feb 2019
+ * @brief	GStreamer plugin to capture sensor data as tensor(s)
+ * @see		http://github.com/nnsuite/nnstreamer
+ * @author	Parichay Kapoor <pk.kapoor@samsung.com>
+ * @bug		No known bugs except for NYI items
+ * @todo  fill in empty functions
+ * @todo  create a sample example and unit tests
+ * @todo  set limit on triggers, channels, buffer capacity, frequency
+ * @todo  verify if multiple triggers can be supported
+ *
+ * This is the plugin to capture data from sensors
+ * and convert them to tensor format.
+ * Current implementation will support accelerators, light and gyro sensors.
+ *
+ */
+
+/**
+ * SECTION:element-tensor_src_iio
+ *
+ * Source element to handle sensors as input.
+ * The output are always in the format of other/tensor or other/tensors.
+ *
+ */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <gst/gstinfo.h>
+#include <gst/gst.h>
+#include <glib.h>
+#include <string.h>
+
+#include "tensor_src_iio.h"
+
+/**
+ * @brief Macro for debug mode.
+ */
+#ifndef DBG
+#define DBG (!self->silent)
+#endif
+
+/**
+ * @brief Macro for debug message.
+ */
+#define silent_debug(...) do { \
+    if (DBG) { \
+      GST_DEBUG_OBJECT (self, __VA_ARGS__); \
+    } \
+  } while (0)
+
+GST_DEBUG_CATEGORY_STATIC (gst_tensor_src_iio_debug);
+#define GST_CAT_DEFAULT gst_tensor_src_iio_debug
+
+/**
+ * @brief tensor_src_iio properties.
+ */
+enum
+{
+  PROP_0,
+  PROP_SILENT,
+  PROP_DEVICE,
+  PROP_TRIGGERS,
+  PROP_VOLTAGE,
+  PROP_CHANNELS,
+  PROP_BUFFER_CAPACITY,
+  PROP_FREQUENCY
+};
+
+/**
+ * @brief Flag to print minimized log.
+ */
+#define DEFAULT_SILENT TRUE
+
+/**
+ * @brief Flag for general default value of string
+ */
+#define DEFAULT_PROP_STRING NULL
+
+/**
+ * @brief Minimum and maximum buffer length for iio
+ */
+#define MIN_BUFFER_CAPACITY 1
+#define MAX_BUFFER_CAPACITY 100
+#define DEFAULT_BUFFER_CAPACITY 1
+
+/**
+ * @brief Minimum and maximum operating frequency for the device
+ * Default frequency chooses the first available frequency supported by device
+ */
+#define MIN_FREQUENCY 1
+#define MAX_FREQUENCY 999999999
+#define DEFAULT_FREQUENCY 0
```

Comment:
Can DEFAULT_FREQUENCY be less then MIN_FREQUENCY? Even thought this default value is dummy, it looks illogical to me.

Reply:
The default frequency is set to 0 indicating that user has not specified any specific frequency and program will use the first value from the list of the available frequency supported.